### PR TITLE
Send entities upon load

### LINKED ISSUE #3696: Entitites turn invisible
Client version:   1.11.2
Server OS:        Linux
Commit id:        05c3e8bed8d1966f5973380b7fc0c033989cb7ba

### Expected behavior
Entities are always visible when they should be.

### Actual behavior
Chests, mobs and pickups randomly turn invisible.

### Steps to reproduce the behavior
Play survival and notice invisible skeletons and zombies attacking you.

### PR DESCRIPTION
**Pull request by Buom01**
These changes where tested with https://github.com/cuberite/cuberite/commit/e290cb4e9b608fbff114205500540ebd324f54a0 and with my armor stand patch https://github.com/Buom01/cuberite

The difference on my case is instant, but I fail to reproduce it with hand spawned mobs.
Now all of my own plugin's mobs are shown instead of ~20% (seem random).
Edit : I will also try @mathiascode world to be sure about the change even if it should be also OK for his world case.

The mystery is: Why some of these mobs were shown while it should be totally hidden since it where probably (or logically) never sent to the client ?
I think that after my pull request some mob could be sent twice.

Additional fix:
- CamelCase on `Client` in the for loop
- Name argument in `cChunkSender::Entity`

Fix #3696

### CODE REVIEW

**Comment by mathiascode** [comment]
Does not seem to work in my world case. Boats are still invisible until you touch the spot they're supposed to be in.

File: src/ChunkSender.h [hunk 1]
```diff
@@ -120,7 +120,7 @@ class cChunkSender:
 	// NOTE that m_BlockData[] is inherited from the cChunkDataCollector
 	unsigned char m_BiomeMap[cChunkDef::Width * cChunkDef::Width];
 	std::vector<Vector3i> m_BlockEntities;  // Coords of the block entities to send
-	// TODO: sEntityIDs    m_Entities;       // Entity-IDs of the entities to send
+	std::vector<UInt32> m_Entities;  // Entity-IDs of the entities to send
```
Comment (madmaxoft) [inline src/ChunkSender.h]: This is named wrong, it's not entities, it's their IDs.

**Comment by mathiascode** [comment]
https://github.com/cuberite/cuberite/blob/master/src/Chunk.cpp#L1729 is the other place in the code where entities are sent to clients, and has the issues that cause entities to be "invisible" (entity spawn is never broadcast to the client).

**Comment by mathiascode** [comment]
@Buom01 Could you test your changes with https://github.com/cuberite/cuberite/blob/master/src/Chunk.cpp#L1729-L1740 removed?

**Comment by Titaniumtown** [comment]
It seems like (according to my testing) that this pr fixes #3696

**Comment by MeMuXin** [comment]
When the entity is actually loaded on the server works excellent, fails sometimes when spawning mob/player

**Comment by Titaniumtown** [comment]
oh, I only tested with another player

**Comment by Titaniumtown** [comment]
@MeMuXin I guess this would only be a partial fix then.

**Comment by Titaniumtown** [comment]
Is this going to be pushed?

**Comment by tigerw** [comment]
hold on

**Comment by Titaniumtown** [comment]
ok

**Comment by tigerw** [comment]
Merged manually.

**Comment by Titaniumtown** [comment]
wdym @tigerw?

**Comment by tigerw** [comment]
Changes merged in #4532 if I'm not mistaken

**Pull request closed without merging**